pkgs/http/test/response_test.dart

```diff
@@ -45,6 +46,23 @@ void main() {
           headers: {'content-type': 'text/plain; charset=iso-8859-1'});
       expect(response.body, equals('föøbãr'));
     });
+    test('test decoding with empty charset if content type is application/json',
+        () {
+      final utf8Bytes = utf8.encode('{"foo":"Привет, мир!"}');
+      var response = http.Response.bytes(utf8Bytes, 200,
+          headers: {'content-type': 'application/json'});
+      expect(response.body, equals('{"foo":"Привет, мир!"}'));
```

Comment:
Ideally, there should be one `except` per test because you are testing a single scenario.

Could you add tests for these additional scenarios:
1. test `.body` with a non-empty charset with content-type application/json data (e.g. with Russian and iso-8859-1)
2. test `.body` with an empty charset with a content-type of test/plain (e.g. with Russian to verify that the default of iso-8859-1 is used)